tools/hermes-parser/js/flow-api-translator/src/flowToFlowDef.js

```diff
@@ -977,6 +978,35 @@ function convertClassDeclaration(
   ];
 }
 
+function convertExpressionToIdOrQualifiedTypeId(
+  node: Expression,
+  context: TranslationContext,
+): DetachedNode<Identifier> | DetachedNode<QualifiedTypeIdentifier> {
+  if (node.type === 'Identifier') {
+    return t.Identifier({name: node.name});
+  } else if (
+    node.type === 'MemberExpression' &&
+    node.property.type === 'Identifier' &&
+    node.object.type !== 'Super'
+  ) {
+    const id = t.Identifier({name: node.property.name});
+    return t.QualifiedTypeIdentifier({
+      qualification: convertExpressionToIdOrQualifiedTypeId(
+        // $FlowFixMe[incompatible-call] conditional already checks for non-Super type
+        node.object,
+        context,
+      ),
+      id,
+    });
+  } else {
```

Comment:
nit: I'd remove this `else` block, since items are returned in each branch above.